src/cli_plugin/install/cleanup.js

```diff
@@ -0,0 +1,32 @@
+import rimraf from 'rimraf';
+import fs from 'fs';
+
+export function cleanPrevious(settings, logger) {
+  return new Promise(function (resolve, reject) {
+    try {
+      fs.statSync(settings.workingPath);
+
+      logger.log('Found previous install attempt. Deleting...');
+      try {
+        rimraf.sync(settings.workingPath);
+      } catch (e) {
+        reject(e);
+      }
+      resolve();
+    } catch (e) {
+      if (e.code !== 'ENOENT') reject(e);
+
+      resolve();
+    }
+  });
+};
+
+export function cleanError(settings) {
+  // delete the working directory.
+  // At this point we're bailing, so swallow any errors on delete.
+  try {
+    rimraf.sync(settings.workingPath);
+    rimraf.sync(settings.plugins[0].path);
```

Comment:
I did not want to restructure all of that code until I knew what the specifications/flow was going to be for installing more than one plugin at a time. Trying to address that refactoring now would lead to coding myself into a corner.